RecoEcal/Configuration/python/RecoEcal_cff.py

```diff
@@ -26,8 +26,14 @@
 from Configuration.Eras.Modifier_pp_on_XeXe_2017_cff import pp_on_XeXe_2017
 
 from RecoEcal.EgammaClusterProducers.islandBasicClusters_cfi import islandBasicClusters
+from RecoHI.HiEgammaAlgos.HiIslandSuperClusters_cfi import islandSuperClusters
+from RecoHI.HiEgammaAlgos.HiCorrectedIslandBarrelSuperClusters_cfi import correctedIslandBarrelSuperClusters
+from RecoHI.HiEgammaAlgos.HiCorrectedIslandEndcapSuperClusters_cfi import correctedIslandEndcapSuperClusters
```

Comment:
all of these 3 imports can collide with the same names possibly already imported.
This ambiguity should be removed.
More transparent way may be to apply era modifiers (.toReplaceWith) to the original definitions used in RecoEcal/EgammaClusterProducers

Reply:
The first option that comes to my mind is to do something like the following. Is that what you are asking for ?
from RecoEcal.EgammaClusterProducers.islandClusteringSequence_cff import islandClusteringSequence
_ecalClustersHI = ecalClusters.copy()
 _ecalClustersHI += islandClusteringSequence

Reply:
that will come as a consequence of the changes I proposed in the original definition location.

Another alternative is to change names of HI-specific islandSuperClusters and other modules to hiIslandSuperClusters and so on.

Reply:
Can you give an example for the proposed changes in the original definition location ? I want to make sure I understand.

Reply:
@ttrk : the point is that (for example) the module "islandSuperClusters" has the same name in both 
​RecoEcal/​EgammaClusterProducers/​python/​islandSuperClusters_cfi.py
and
RecoHI/​HiEgammaAlgos/​python/​HiIslandSuperClusters_cfi.py
(and a different configuration).

Therefore, if you import one of them in this RecoEcal_cff.py you prevent importing the other one, or you end up with ambiguities in the configurations.

The solution could be to rename, e.g., hiIslandSuperClusters the module in HiIslandSuperClusters_cfi.py (and the same for the other modules), and then use the eras to add the correct modules either here (as you do now), or, probably better, inside the islandClusteringSequence, as you also suggested.

Reply:
@perrotta Thanks for the explanation. I understand the ambiguity resulting from that the modules in different files are given the same names.

However when @slava77 mentioned it as "Another alternative" (https://github.com/cms-sw/cmssw/pull/20929#discussion_r145709059), it sounded to me as if renaming the HI-specific modules is an "alternative" way and one can solve the ambiguity without renaming modules.

Anyway I will rename the HI-specific modules.